src/codec/rle.rs

```diff
@@ -132,10 +168,10 @@ pub fn rle_32_decompress(
         )));
     }
 
-    process_plane(&mut input_cursor, width, height, &mut output[3..])?;
-    process_plane(&mut input_cursor, width, height, &mut output[2..])?;
-    process_plane(&mut input_cursor, width, height, &mut output[1..])?;
-    process_plane(&mut input_cursor, width, height, &mut output[0..])?;
+    process_plane(&mut input_cursor, width, height, &mut output[3..])?; // alpha
+    process_plane(&mut input_cursor, width, height, &mut output[2..])?; // blue
+    process_plane(&mut input_cursor, width, height, &mut output[1..])?; // green
+    process_plane(&mut input_cursor, width, height, &mut output[0..])?; // red
```

Comment:
I'm just trying to understand this, so please correct me if I'm wrong.
The indexes in the output slice are starting positions for color channels in the output slice, right? And then in the process plane function when we process the data and want to save it to the output we have to multiply position by 4 since there are 4 channels?

I wonder if it would be easier to understand if we process all channels at the same time in one function, but then it would complicate the process plane function even more 😄

Reply:
Correct. Took me a while to figure that out too. It's basically the position of the first byte for that color in the first pixel, and then we write the next pixel 4 bytes from there, etc.